cli/azd/pkg/ext/hooks_manager.go

```diff
@@ -81,6 +82,10 @@ func (h *HooksManager) filterConfigs(
 		}
 
 		for _, hook := range hooks {
+			if hook == nil {
```

Comment:
This is great! I'm wondering how we can verify in all other places where `*HookConfig` is used, are we guaranteed that they're non-nil in all those use cases?